js/ui/appDisplay.js

```diff
@@ -1772,16 +1772,18 @@ const FolderView = new Lang.Class({
             if (!this.iconsNeedRedraw())
                 return;
 
-            this._redisplay();
-
-            this._folderIcon.icon.reloadIcon();
+            if (!this._recreateIconWorkId)
+                this._recreateIconWorkId = Main.initializeDeferredWork(this._folderIcon.actor, Lang.bind(this, this._redisplay));
+            else
+                Main.queueDeferredWork(this._recreateIconWorkId);
```

Comment:
Is this really needed now that 9d63828 has been reverted? https://github.com/endlessm/gnome-shell/commit/d5595f5fa1aa1a869426dcb4e3fff3c4869bcafb

Reply:
Actually, this seems wrong and behind the crash I've been seeing: when the folder changes, the workID created for the actor showing the previous state of the folder is invalid, since such actor has been destroyed by the time this is called.

Reply:
Indeed, this change is behind the backtrace I've been seeing that breaks DnD after the first DnD operation. Dropping this commit seems fine in my tests so far, and I can'te see any problem with a "broken grab", but there's not much extra info in the commit message that clarifies what exactly that is about, so I'm not 100% sure.